Std/Data/Int/DivMod.lean

```diff
@@ -864,6 +878,174 @@ theorem eq_one_of_mul_eq_one_right {a b : Int} (H : 0 ≤ a) (H' : a * b = 1) :
 theorem eq_one_of_mul_eq_one_left {a b : Int} (H : 0 ≤ b) (H' : a * b = 1) : b = 1 :=
   eq_one_of_mul_eq_one_right H <| by rw [Int.mul_comm, H']
 
+/-!
+# `bmod` ("balanced" mod)
+
+We use balanced mod in the omega algorithm,
+to make ±1 coefficients appear in equations without them.
+-/
+
+/--
+Balanced mod, taking values in the range [- m/2, (m - 1)/2].
+-/
+def bmod (x : Int) (m : Nat) : Int :=
+  let r := x % m
+  if r < (m + 1) / 2 then
+    r
+  else
+    r - m
```

Comment:
Is this better or worse as:
```suggestion
  let offset := m / 2; (x + offset) % m - offset
```

The fact it avoids a branch means it might be both faster and easier to prove things about

Reply:
Oooh... That is probably much better. :-)

Reply:
I had a bit of a look. Some of the inequalities do get easier, but the annoying lemma at the end `bmod_natAbs_plus_one`, looks more annoying: the proof I have in mind splits into cases along the same lines as the definition here.

I'll keep this suggestion in mind, but it is a low priority to replace the proofs for now.